valkyrie/README.md

```diff
@@ -1,30 +1,101 @@
 # Valkyrie
 
-Welcome to your new gem! In this directory, you'll find the files you need to be able to package up your Ruby library into a gem. Put your Ruby code in the file `lib/sleipnir`. To experiment with that code, run `bin/console` for an interactive prompt.
+Valkyrie is a gem for enabling multiple backends for storage of files and metadata in Samvera,
+currently developed in a subdirectory of the [proof-of-concept app of the same name](..).
+
+[![CircleCI](https://circleci.com/gh/samvera-labs/valkyrie.svg?style=svg)](https://circleci.com/gh/samvera-labs/valkyrie)
+[![Coverage Status](https://coveralls.io/repos/github/samvera-labs/valkyrie/badge.svg?branch=master)](https://coveralls.io/github/samvera-labs/valkyrie?branch=master)
+[![Stories in Ready](https://badge.waffle.io/samvera-labs/valkyrie.png?label=ready&title=Ready)](https://waffle.io/samvera-labs/valkyrie)
 
-TODO: Delete this and the text above, and describe your gem
 
 ## Installation
 
 Add this line to your application's Gemfile:
 
 ```ruby
-gem 'valkyrie'
+gem 'valkyrie', github: 'samvera-labs/valkyrie'
 ```
 
 And then execute:
 
     $ bundle
 
-Or install it yourself as:
 
-    $ gem install valkyrie
+## Configuration
+
+Valkyrie is configured in `config/valkyrie.yml`.  For each environment you must set
+two values.  The first, `adapter`, is the store where Valkyrie will put the metadata.
+The valid values for `adapter` are `:postgres` (default), `:fedora`(actually ActiveFedora),
+`:memory` (should only be used for running tests, because it's not persistent).
+You can register your own adapters in `config/initializers/valkyrie.rb`.
+
+Valkyrie will automatically persist to the metadata store you have chosen and write
+to Solr synchronously (using the `:indexing_persister`).
+
+The second configuration value `storage_adapter` is where the binaries are stored.
+The valid values for `storage_adapter` are `:disk` and `:memory`. `:memory` should
```

Comment:
Again, any value is valid here as long as it's registered in `Valkyrie::StorageAdapter#register`